docs/user-guide/03-convenience-features.md

```diff
@@ -383,6 +383,70 @@ int caller()
 }
 ```
 
+---
+layout: user-guide
+permalink: /user-guide/convenience-features-if-let
+---
+## `if_let` syntax
+Slang supports `if (let ...)` syntax to simplify the code when checking the type of a value. The syntax is similar to Rust's
+`if let` syntax, the variable defined by `let` must be an `Optional<T>` type, for example:
+
+```csharp
+interface IFoo
+{
+    void foo();
+}
+
+struct MyImpl1 : IFoo
+{
+    void foo() { printf("MyImpl1");}
+}
+
+struct MyImpl2 : IFoo
+{
+    void foo() { printf("MyImpl2");}
+}
+
+struct MyImpl3 : IFoo
+{
+    void foo() { printf("MyImpl3");}
+}
+
+void test(IFoo foo)
```

Comment:
The example here should not use `as` operator since we haven't talked about it at this point of the documentation.

We can simply use:
```
Optional<int> getOptInt() { ... }

void test()
{
    if (let x = getOptInt())
    {
          // if we are here, `getOptInt` returns a value `int`.
          // and `x` represents the `int` value.
    }
}
```

Reply:
act